agent/exec/containerd/executor.go

```diff
@@ -0,0 +1,107 @@
+package containerd
+
+import (
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/docker/containerd/api/services/execution"
+
+	"github.com/docker/swarmkit/agent/exec"
+	"github.com/docker/swarmkit/agent/secrets"
+	"github.com/docker/swarmkit/api"
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc"
+	//"google.golang.org/grpc/grpclog"
+)
+
+type executor struct {
+	client       execution.ExecutionServiceClient
+	secrets      exec.SecretsManager
+	bundleDir    string
+	contentStore string
+}
+
+var _ exec.Executor = &executor{}
+
+// Lifted from github.com/docker/containerd/cmd/ctr/utils.go
```

Comment:
I wrote this last night, as well. Maybe, we need to provide some easy client making tools there.